test/Common/DurableTaskEndToEndTests.cs

```diff
@@ -1474,6 +1474,53 @@ public async Task TerminateOrchestration(bool extendedSessions, string storagePr
             }
         }
 
+        /// <summary>
+        /// End-to-end test which validates the Suspend-Resume functionality.
+        /// </summary>
+        [Theory]
+        [Trait("Category", PlatformSpecificHelpers.TestCategory)]
+        [MemberData(nameof(TestDataGenerator.GetBooleanAndFullFeaturedStorageProviderOptions), MemberType = typeof(TestDataGenerator))]
+        public async Task SuspendResumeOrchestration(bool extendedSessions, string storageProvider)
+        {
+            string[] orchestratorFunctionNames =
+            {
+                nameof(TestOrchestrations.Counter),
+            };
+
+            using (ITestHost host = TestHelpers.GetJobHost(
+                this.loggerProvider,
+                nameof(this.SuspendResumeOrchestration),
+                extendedSessions,
+                storageProviderType: storageProvider))
+            {
+                await host.StartAsync();
+                var client = await host.StartOrchestratorAsync(orchestratorFunctionNames[0], 0, this.output);
+                await client.WaitForStartupAsync(this.output);
+
+                // Test case 1: Suspend changes the status Running->Suspended
+                await client.SuspendAsync("sleepyOrch");
+                await Task.Delay(2000);

```

Comment:
Consider using a polling while loop instead of sleeping to make the test code more reliable.

Reply:
Good idea! thanks